plugins/rts54xx/fu-plugin-rts54xx.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin.h"
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-rts54xx-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRES_QUIRK, FU_QUIRKS_PLUGIN);
+}
+
+gboolean
+fu_plugin_update (FuPlugin *plugin,
+		  FuDevice *device,
+		  GBytes *blob_fw,
+		  FwupdInstallFlags flags,
+		  GError **error)
+{
+	g_autoptr(FuDeviceLocker) locker = NULL;
+
+	/* check */
+	if (g_bytes_get_size (blob_fw) % 0x10000 != 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "File not expected size, expected multiple of 0x%04x",
+			     (guint) 0x10000);
+		return FALSE;
+	}
+
+	/* write */
+	locker = fu_device_locker_new (device, error);
+	if (locker == NULL)
+		return FALSE;
+	return fu_device_write_firmware (device, blob_fw, error);
```

Comment:
I _think_ it's worth special casing the VID/PID you are initially adding in this PR with an extra quirk setting (something like `needs-force-update` maybe?)  Since that's the reference VID/PID you have no "guaranteed" way to distinguish incompatible firmware changes in designs that used that versus designs using reference firmware in production hardware.

Reply:
(Once you have I2C devices you're talking to, that might of course change though and you'll have new ways to tell what kind of device you really have behind the bridge)

Reply:
I've worked a bit on this over the weekend; see https://github.com/hughsie/fwupd/pull/690 for one idea

Reply:
I guess you need a flag now for this?

Reply:
Flag, pssh! I've got a whole new derived type! :)